File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Transport/SeaTransport.java
package com.logistics.Components.Transport;

public class SeaTransport extends Transport {
    private String shipName;

    public SeaTransport() {
        super();
        shipName = "";
    }

    public SeaTransport(int code, int speed, int costPerMeter, String route, String shipName) {
        super(code, speed, costPerMeter, route);
        this.shipName = shipName;
    }

    // Copy Constructor
    public SeaTransport(SeaTransport other) {
        super(other);
        this.shipName = other.shipName;
    }

    // Clone method
    @Override
    public SeaTransport clone() {
        return new SeaTransport(this);
    }

    // Getters
    public String getShipName() {
        return shipName;
    }

    // Setters
    public void setShipName(String shipName) {
        this.shipName = shipName;
    }

    @Override
    public String to_string() {
        return super.to_string() +
                "\nShip Name: " + shipName;
    }
}

File: Task 1/C++/Components/Transport/SeaTransport.cpp
#include "SeaTransport.h"

SeaTransport::SeaTransport() : Transport(){
    shipName = "";
};

SeaTransport::SeaTransport(const int code, float speed, float costPerMeter, const std::string route, const std::string shipName) 
            : Transport(code, speed, costPerMeter, route){
    this->shipName = shipName;
}

SeaTransport::SeaTransport(const SeaTransport& other) : Transport(other) {
    this->shipName = other.shipName;
}

std::string SeaTransport::to_string(){
    return Transport::to_string()+
            "\nShip Name: "+shipName;
}